### PR TITLE
Show delete confirmation for the action when saving empty comment

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ https://github.com/Expensify/App/issues/7146

### Tests | QA Steps

1. Open the app and navigate to chat
2. User opens edit box to edit a message they've already sent (via cmd+k or pencil2
3. User deletes all text in box then hit hits enter/return or clicks Save

- [x] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web | Desktop

https://user-images.githubusercontent.com/24370807/152855537-d3b70c9e-00a3-4367-b339-1d6d252a14fa.mp4

#### Mobile Web

https://user-images.githubusercontent.com/24370807/152861337-7cb475cf-df9b-451e-9a1b-c654fda7480e.mp4

#### Android

https://user-images.githubusercontent.com/24370807/152861147-0a354496-ff54-4ce6-8e3a-c3e2e8ccfe7a.mp4

### CODE REVIEW

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -1232,6 +1243,9 @@ function deleteReportComment(reportID, reportAction) {
         ],
     };
 
+    // Clear the draft comment for action if any.
+    saveReportActionDraft(reportID, reportAction.reportActionID, '');
+
```
Comment (parasharrajat) [inline src/libs/actions/Report.js:1248]: To answer your question from the issue, showdeleteModal does not take any callback for now so we can't hook this call to the deleteModal. But I think it is fairly safe to call this method here. But let me know if you think this is confusing.

**Review by stitesExpensify — COMMENTED** [review]
Code looks good, while testing I noticed that if you cancel out of the default confirmation, it focuses the main input instead of the comment you're editing

**Comment by parasharrajat** [comment]
I see. Yeah, due to focus being automatically set on the composer when modal is closed. But to solve this. I will have to pass a callback to showDeleteModal and run that to manage focus. Also, I will have to disable the autofocus from the deleteModal. I will update the PR.

**Comment by stitesExpensify** [comment]
Bump :)

**Comment by parasharrajat** [comment]
Thanks for the bump. I will finish this today.

**Comment by parasharrajat** [comment]
Working on it. I am struggling to set the focus on the draft editor.

**Comment by parasharrajat** [comment]
Ready, updating screens shortly...

File: src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js [hunk 2]
```diff
@@ -278,8 +298,10 @@ class PopoverReportActionContextMenu extends React.Component {
                 <ConfirmModal
                     title={this.props.translate('reportActionContextMenu.deleteComment')}
                     isVisible={this.state.isDeleteCommentConfirmModalVisible}
+                    shouldSetModalVisibility={this.state.shouldSetModalVisibilityForDeleteConfrmation}
```
Comment (stitesExpensify) [inline src/pages/home/report/ContextMenu/PopoverReportActionContextMenu.js:301]: typo: `confrmation`

**Review by stitesExpensify — APPROVED** [review]

**Comment by stitesExpensify** [comment]
Sorry for the delay on the review. LGTM!

**Comment by Stutikuls** [comment]
Title- [High] Keyboard : Controls(Edit, Delete, etc.) are not appearing using keyboard.
Actual -Controls(Edit, Delete, etc.) are not appearing using keyboard, displayed on hover.
Expected - Controls(Edit, Delete, etc.) should appear using keyboard.
WCAG failure - 1.4.13
Reproducible in staging?: - Yes
Version Number: - v1.1.39-3
Platforms - Web, Desktop, Mobile-web, iOS

https://user-images.githubusercontent.com/97600950/156757522-8f894bd9-02bc-42e9-92c8-aca9521120a3.mp4

**Note -** Due to installation of build not able to check on Android.

File: src/pages/home/report/ReportActionItemMessageEdit.js [hunk 3]
```diff
@@ -122,6 +123,18 @@ class ReportActionItemMessageEdit extends React.Component {
         this.debouncedSaveDraft.cancel();
 
         const trimmedNewDraft = this.state.draft.trim();
+
+        // When user tries to save the empty message, it will delete it. Prompt the user to confirm deleting.
+        if (!trimmedNewDraft) {
+            ReportActionContextMenu.showDeleteModal(
+                this.props.reportID,
+                this.props.action,
+                false,
```
Comment (situchan) [inline src/pages/home/report/ReportActionItemMessageEdit.js:132]: This should have been `true`. Later this caused minor focus issue - [Composer doesn't get focused automatically after delete message in edit mode](https://github.com/Expensify/App/issues/30135)

**Pull request merged**